kubectl-openebs/cli/command/describe/pool_info.go

```diff
@@ -0,0 +1,177 @@
+			if namespace, _ = cmd.Flags().GetString("namespace"); namespace == "" {
+				// NOTE: The error comes as nil even when the ns flag is not specified
+				namespace = "openebs"
+			}
+			util.CheckErr(RunPoolInfo(cmd, args, namespace), util.Fatal)
+		},
+	}
+	return cmd
+}
+
+// RunPoolInfo method runs info command and make call to DisplayPoolInfo to display the results
+func RunPoolInfo(cmd *cobra.Command, pools []string, ns string) error {
+	if len(pools) != 1 {
+		return errors.New("Please give one cspi name to describe")
+	}
+
+	clientset, err := client.NewK8sClient(ns)
+	if err != nil {
+		return errors.Wrap(err, "Failed to execute pool info command")
+	}
+
+	// Fetch the CSPI object by passing the name of CSPI taken through CLI in ns namespace
+	poolName := pools[0]
+	poolInfo, err := clientset.GetcStorPool(poolName)
+	if err != nil {
+		return errors.Wrap(err, "Error getting pool info")
+	}
+
+	poolDetails := util.PoolInfo{
+		Name:           poolInfo.Name,
+		HostName:       poolInfo.Spec.HostName,
+		Size:           poolInfo.Status.Capacity.Total.String(),
+		FreeCapacity:   poolInfo.Status.Capacity.Free.String(),
+		ReadOnlyStatus: poolInfo.Status.ReadOnly,
+		Status:         poolInfo.Status.Phase,
+		RaidType:       poolInfo.Spec.PoolConfig.DataRaidGroupType,
+	}
+	// Fetch all the raid groups in the CSPI
+	RaidGroupsInPool := poolInfo.GetAllRaidGroups()
+
+	// Fetch all the block devices in the raid groups associated to the CSPI
+	var BlockDevicesInPool []string
+	for _, item := range RaidGroupsInPool {
+		BlockDevicesInPool = append(BlockDevicesInPool, item.GetBlockDevices()...)
+	}
+
+	// Fetch info for every block device
+	var BdInfo []util.BlockDevicesInfoInPool
+	for _, item := range BlockDevicesInPool {
+		bd, err := clientset.GetBlockDevice(item)
+		if err != nil {
+			return errors.Wrap(err, "Error getting block device info")
+		}
+
+		BdInfo = append(BdInfo, util.BlockDevicesInfoInPool{
+			Name:     bd.Name,
+			Capacity: bd.Spec.Capacity.Storage,
+			State:    bd.Status.State,
+		})
+	}
+
+	// Fetch info for provisional replica
+	CVRsInPool, err := clientset.GetCVRByPoolName(poolName)
+	if err != nil {
+		return errors.Wrap(err, "Error getting block device info")
+	}
+
+	var CVRInfoInPool []util.CVRInfo
+	for _, cvr := range CVRsInPool.Items {
+		CVRInfoInPool = append(CVRInfoInPool, util.CVRInfo{
+			Name:    cvr.Name,
+			PvcName: clientset.GetPVCNameByCVR(cvr.Labels["openebs.io/persistent-volume"]),
+			Size:    cvr.Status.Capacity.Total,
+			Status:  cvr.Status.Phase,
+		})
+	}
+
+	// Printing the filled details of the Pool
+	err = util.PrintByTemplate("pool", cStorPoolInstanceInfoTemplate, poolDetails)
+	if err != nil {
+		return err
+	}
+
+	for _, bd := range BdInfo {
+		err = util.PrintByTemplate("bd", blockDevicesInfoFromCSPI, bd)
+		if err != nil {
+			return err
+		}
+	}
+
+	for _, cvr := range CVRInfoInPool {
+		err = util.PrintByTemplate("cvr", provisionedReplicasInfoFromCSPI, cvr)
```

Comment:
- `PrintByTemplate` may not be a good UI/UX as in case of multiple CVRs the above will generate something like this:-
```
Provisioned Replicas Details :
----------------
Name     : pvc-b84f60ae-3f26-4110-a85d-bce7ec00dacc-default-cstor-disk-dcrm
PVC Name : mongo
Size     : 1.04G
Status	 : Healthy
Provisioned Replicas Details :
----------------
Name     : pvc-b84f60ae-3f26-4110-a85d-bce7ec00dacc-default-cstor-disk-dcrm
PVC Name : mongo
Size     : 1.04G
Status	 : Healthy

```
The above should have been like this, similar to `kubectl get` list.
```
Provisioned Replicas Details :
----------------
Name   PvcName  Size    Status
cvr-1  pvc-1    107641  Healthy 
cvr-2  pvc-2    187261  Degraded
```
You can refer this code for the same:-
https://github.com/openebs/openebsctl/blob/de75453658baa6be23d739f3198df012e7c35f32/kubectl-openebs/cli/command/describe/pvc_info.go#L213